libs/jsonforms-components/src/lib/Controls/AddressLookup/AddressLookUpControl.tsx

```diff
@@ -112,40 +119,122 @@ export const AddressLookUpControl = (props: AddressLookUpProps): JSX.Element =>
+          });
+        }
+      }
+    }, [selectedIndex, open]);
+
+/* istanbul ignore next */
+const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'ArrowDown') {
+        e.preventDefault();
+        setSelectedIndex((prevIndex) =>
+            prevIndex < suggestions.length - 1 ? prevIndex + 1 : 0
+        );
+        handleDropdownChange(e.currentTarget.value)
+    } else if (e.key === 'ArrowUp') {
+        e.preventDefault();
+        setSelectedIndex((prevIndex) =>
+            prevIndex > 0 ? prevIndex - 1 : suggestions.length - 1
+        );
+        handleDropdownChange(e.currentTarget.value)
+    } else if (e.key === 'Enter') {
+      handleDropdownChange(e.currentTarget.value)
+      setLoading(false);
+      if (selectedIndex >= 0) {
+        document.getElementById("goaInput")?.blur()
+        const suggestion = suggestions[selectedIndex];
+        if(suggestion){
+          setTimeout(() => {
+              handleSuggestionClick(suggestion);
+              setOpen(false);
+            }, 100);
+          }
+        }
+    }
+};
+
   return (
     <div>
       {renderHelp()}
       <GoAFormItem label={label} error={errors?.['addressLine1'] ?? ''} data-testId="form-address-line1">
-        <SearchBox>
+        <SearchBox
+            onKeyDown={(e) => {
+              if(open){
+                handleKeyDown(e)
+              }
+            }
+          }
+        >
           <GoAInput
+            id="goaInput"
             leadingIcon={autocompletion ? 'search' : undefined}
             name="addressLine1"
             testId="address-form-address1"
             ariaLabel={'address-form-address1'}
             placeholder="Start typing the first line of your address, required."
             value={address?.addressLine1 || ''}
-            onChange={(name, value) => handleDropdownChange(value)}
+            onChange={(value) => handleDropdownChange(value)}
             onBlur={(name) => handleRequiredFieldBlur(name)}
             width="100%"
           />
-          {loading && autocompletion && <GoASkeleton type="text" data-testId="loading" key={1} />}
+          {loading && autocompletion && <GoACircularProgress variant="inline" size="small" visible={true}></GoACircularProgress> }
+
           {suggestions && autocompletion && (
-            <ul className="suggestions" tabIndex={0}>
+            <ul ref={dropdownRef} className="suggestions" tabIndex={0}>
               {suggestions &&
                 autocompletion &&
                 open &&
                 suggestions.map((suggestion, index) => (
-                  <li key={index} onClick={(e) => handleSuggestionClick(suggestion)}>
-                    {`${suggestion.Text}  ${suggestion.Description}`}
+                  <li
+                  data-index={index}
+                  key={index}
+                  onClick={() => {
+                    handleSuggestionClick(suggestion)
+                  }}
+                  style={{
```

Comment:
style part can be put in style-components.tsx file. it can be handle by li state.